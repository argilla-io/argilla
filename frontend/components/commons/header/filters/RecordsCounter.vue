<template>
  <p class="records-counter" v-if="total > 0">
    Records ({{ total | formatNumber }})
  </p>
</template>

<script>
export default {
  props: {
    total: {
      type: Number,
      required: true,
    },
  },
};
</script>

<style lang="scss" scoped>
.records-counter {
  color: palette(grey, medium);
  margin-left: auto;
  margin-right: 0;
  display: block;
  text-align: right;
}
</style>
