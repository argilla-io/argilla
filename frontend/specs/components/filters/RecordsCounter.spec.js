import { mount } from "@vue/test-utils";
import RecordsCounter from "@/components/commons/header/filters/RecordsCounter";

function mountRecordsCounter() {
  return mount(RecordsCounter, {
    propsData: {
      dataset: {
        task: "TextClassification",
        results: {
          total: 400,
        },
        query: {
          annotated_as: undefined,
          annotated_by: undefined,
          from: undefined,
          limit: undefined,
          metadata: {},
          predicted: undefined,
          predicted_as: ["Alta"],
          predicted_by: undefined,
          score: undefined,
          status: undefined,
          text: "de",
        },
      },
      showWhenFiltered: true,
      excludedFilter: ["text"],
    },
  });
}

describe("RecordsCounter", () => {
  let spy = jest.spyOn(console, "error");
  afterEach(() => spy.mockReset());

  test("renders properly", () => {
    const wrapper = mountRecordsCounter();
    expect(wrapper).toMatchSnapshot();
  });
});
