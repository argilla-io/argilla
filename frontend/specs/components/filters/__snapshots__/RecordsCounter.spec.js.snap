// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordsCounter renders properly 1`] = ``;
